src/api/error.ts

```diff
@@ -1,23 +1,25 @@
-// @flow
-
 import { log } from "../../react-native-logs.config";
 
 const logger = log.extend( "INatApiError" );
 
 export class INatApiError extends Error {
   // Object literal of the JSON body returned by the server
-  json: Object;
+  json: Record<string, unknown>;
 
   // HTTP status code of the server response
   status: number;
 
   // Additional context information
-  context: ?Object;
+  context: Record<string, unknown> | null;
 
-  constructor( json: Object, status?: number, context?: Object ) {
+  constructor(
+    json: Record<string, unknown> & { status: string | number },
+    status?: number,
+    context?: Record<string, unknown> | null
+  ) {
     super( JSON.stringify( json ) );
     this.json = json;
-    this.status = status || json.status;
+    this.status = status || +json.status;
```

Comment:
Oh nice, great fix. I checked other places, but yeah, afaict, confirmed that we always treat these as numbers: 

<img width="607" height="902" alt="Image" src="https://github.com/user-attachments/assets/54eec9a0-f500-4b45-9a4a-4ed4f0cab958" />

nit: I think the `+` operator coercion can be hard to notice and the codebase tends to use the `Number` constructor.

```suggestion
    this.status = status || Number(json.status);
```

Reply:
Hmm I wonder how those checks were working in the past. Was there some string to number conversion happening elsewhere? IIRC, all error response statuses from the API return strings